tests/Aspire.Hosting.Garnet.Tests/GarnetFunctionalTests.cs

```diff
@@ -59,4 +55,158 @@ await pipeline.ExecuteAsync(async token =>
 
          }, cts.Token);
     }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    [RequiresDocker]
+    public async Task WithDataShouldPersistStateBetweenUsages(bool useVolume)
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(5));
+        var pipeline = new ResiliencePipelineBuilder()
+                            .AddRetry(new() { MaxRetryAttempts = 10, Delay = TimeSpan.FromSeconds(10) })
+                            .Build();
+        string? volumeName = null;
+        string? bindMountPath = null;
+
+        try
+        {
+            var builder1 = TestDistributedApplicationBuilder.Create().WithTestAndResourceLogging(testOutputHelper);
+            var garnet1 = builder1.AddGarnet("garnet");
+
+            if (useVolume)
+            {
+                // Use a deterministic volume name to prevent them from exhausting the machines if deletion fails
+                volumeName = VolumeNameGenerator.CreateVolumeName(garnet1, nameof(WithDataShouldPersistStateBetweenUsages));
+
+                // if the volume already exists (because of a crashing previous run), try to delete it
+                DockerUtils.AttemptDeleteDockerVolume(volumeName);
+                garnet1.WithDataVolume(volumeName);
+            }
+            else
+            {
+                bindMountPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+
+                if (!Directory.Exists(bindMountPath))
+                {
+                    if (OperatingSystem.IsWindows())
+                    {
+                        Directory.CreateDirectory(bindMountPath);
+                    }
+                    else
+                    {
+                        Directory.CreateDirectory(bindMountPath, UnixFileMode.UserRead | UnixFileMode.UserWrite | UnixFileMode.GroupRead | UnixFileMode.OtherRead);
```

Comment:
Maybe a comment on why we need to use this overload.

Reply:
There is issue in bind mount for linux env, i saw access denied in container log but it seems this overload not helping us. This test still fails on linux helix.

Reply:
I've added a comment and change the permissions to "Other" read/write.